tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java

```diff
@@ -929,37 +933,78 @@ public void TestContinuationOrder() {
     static final Set<String> NOT_UK = ImmutableSet.of(
         "therm-us", "bushel", "barrel");
 
-    public void TestSystems() {
-        Multimap<String, String> toSystems = converter.getSourceToSystems();
+    public static final Set<String> OTHER_SYSTEM = ImmutableSet.of(
+        "g-force", "dalton", "calorie", "earth-radius",
+        "solar-radius", "solar-radius", "astronomical-unit", "light-year", "parsec", "earth-mass",
+        "solar-mass", "bit", "byte", "karat", "solar-luminosity", "ofhg", "atmosphere",
+        "pixel", "dot", "permillion", "permyriad", "permille", "percent", "karat", "portion",
+        "minute", "hour", "day", "day-person", "week", "week-person",
+        "year", "year-person", "decade", "month", "month-person", "century",
+        "arc-second", "arc-minute", "degree", "radian", "revolution",
+        "electronvolt",
+        // quasi-metric
+        "dunam", "mile-scandinavian", "carat", "cup-metric", "pint-metric"
+        );
 
-        Map<String, TargetInfo> data = converter.getInternalConversionData();
-        for (Entry<String, TargetInfo> entry : data.entrySet()) {
-            String unit = entry.getKey();
-            TargetInfo value = entry.getValue();
-            String inputFactor = value.inputParameters.get("factor");
-            if (inputFactor == null) {
-                inputFactor = "";
-            }
-            boolean usSystem = !NOT_US.contains(unit) &&
-                (OK_BOTH.contains(unit)
-                    || OK_US.contains(unit)
-                    || usSystemPattern.matcher(inputFactor).find());
-
-            boolean ukSystem = !NOT_UK.contains(unit) &&
-                (OK_BOTH.contains(unit)
-                    || OK_UK.contains(unit)
-                    || ukSystemPattern.matcher(inputFactor).find());
-
-            Collection<String> systems = toSystems.get(unit);
-            if (systems == null) {
-                systems = Collections.emptySet();
-            }
-            if (!assertEquals(unit + ": US? (" + inputFactor + ")", usSystem, systems.contains("ussystem"))) {
-                int debug = 0;
+    public void TestSystems() {
+//        Map<String, TargetInfo> data = converter.getInternalConversionData();
+        Output<String> metricUnit = new Output<>();
+        Multimap<Set<UnitSystem>, R3<String, ConversionInfo, String>> systemsToUnits = TreeMultimap.create(Comparators.lexicographical(Ordering.natural()), Ordering.natural());
+        for (String longUnit : VALID_REGULAR_UNITS) {
+            String unit = Units.getShort(longUnit);
+            if (unit.equals("generic")) {
+                continue;
             }
-            if (!assertEquals(unit + ": UK? (" + inputFactor + ")", ukSystem, systems.contains("uksystem"))) {
+            if (unit.contentEquals("centiliter")) {
                 int debug = 0;
```

Comment:
I'm curious what this "centiliter? then debug=0" is about? (I see code style isn't to have lots of explanatory comments though.)

Reply:
Those are just debug statements; there is a separate ticket to clean those up.